MASStorage/Classes/_private_/MASCloudStorage.m

```diff
@@ -26,6 +27,21 @@ + (void)findObjectUsingKey:(NSString *)key
     NSString *pathURL;
     NSString *storageEndPoint = [[[MASConfiguration currentConfiguration] endpointPathForKey:MASStorageEndPointKey] stringByDeletingLastPathComponent];
 
+    //
+    // Endpoint validation
+    //
+    if (!storageEndPoint || [storageEndPoint isKindOfClass:[NSString class]] || storageEndPoint.length == 0)
```

Comment:
Don't you want to be checking if ![storageEndPoint isKindOfClass:[NSString class]]?
In other words, if the storage endpoint is NOT a string?